controller/controller.py

```diff
@@ -556,15 +556,22 @@ async def record_with_values(
         topic: str,
         *,
         record_type: Optional[Type[T]] = None,
+        timeout: float = 5.0,  # seconds
         **values,
     ) -> Union[T, Mapping[str, Any]]:
         """Get a record from an event with values matching those passed in."""
         try:
-            event = await self.event_queue.get(
-                lambda event: event.topic == topic
-                and all(
-                    [event.payload.get(key) == value for key, value in values.items()]
-                )
+            event = await asyncio.wait_for(
+                self.event_queue.get(
```

Comment:
This is really close! As it turns out, the event_queue.get method takes a timeout parameter so we can use that instead of the asyncio.wait_for in this method. (Under the hood, this is exactly what the event_queue.get method is doing so the methodology is good!)